### ISSUE TITLE
Node.js install per documentation fails with latest node

### ISSUE THREAD

**Issue by Martin-Pitt** [grpc/grpc#9418]
TL;DR: v1.0.0 doesn't compile on Node v7 as it's missing third_party source files. Following quickstart instructions results in confused Node.js users. Recommend updating docs.

---

I'm using Node v7.4.0 / macOS 10.11.6 and tried to follow the guide to get started on GRPC w/ Node.
( http://www.grpc.io/docs/quickstart/node.html )

(Logs amended for some brevity and privacy)
I encountered the following issues:
```
$ git clone -b v1.0.0 https://github.com/grpc/grpc grpc
Cloning into 'grpc'...
$ npm i
…
node-pre-gyp ERR! Tried to download(404): https://storage.googleapis.com/grpc-precompiled-binaries/node/grpc/v1.0.0/node-v51-darwin-x64.tar.gz
node-pre-gyp ERR! Pre-built binaries not found for grpc@1.0.0 and node@7.4.0 (node-v51 ABI) (falling back to source compile with node-gyp)
…
(starts compiling with xcodebuild)
…
  CC(target) Release/obj.target/grpc/src/core/ext/transport/chttp2/server/insecure/server_chttp2.o
  CC(target) Release/obj.target/grpc/src/core/ext/transport/chttp2/server/insecure/server_chttp2_posix.o
  CC(target) Release/obj.target/grpc/src/core/ext/transport/chttp2/client/insecure/channel_create.o
  CC(target) Release/obj.target/grpc/src/core/ext/transport/chttp2/client/insecure/channel_create_posix.o
  CC(target) Release/obj.target/grpc/src/core/ext/lb_policy/grpclb/load_balancer_api.o
In file included from ../src/core/ext/lb_policy/grpclb/load_balancer_api.c:34:
In file included from .././src/core/ext/lb_policy/grpclb/load_balancer_api.h:40:
.././src/core/ext/lb_policy/grpclb/proto/grpc/lb/v1/load_balancer.pb.h:38:10: fatal error: 'third_party/nanopb/pb.h' file not found
#include "third_party/nanopb/pb.h"
         ^
1 error generated.
make: *** [Release/obj.target/grpc/src/core/ext/lb_policy/grpclb/load_balancer_api.o] Error 1
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
```

Looking into my filesystem, yep `third_party/nanopb/` is an empty folder.
Looking into this GitHub on the master branch, I see there ought to be a bunch of things.

However, after checkout of master branch and running `npm i`:
```
$ npm i

> grpc@1.1.0-dev install /…/grpc
> node-pre-gyp install --fallback-to-build

node-pre-gyp ERR! Tried to download(404): https://storage.googleapis.com/grpc-precompiled-binaries/node/grpc/v1.1.0-dev/node-v51-darwin-x64.tar.gz
node-pre-gyp ERR! Pre-built binaries not found for grpc@1.1.0-dev and node@7.4.0 (node-v51 ABI) (falling back to source compile with node-gyp)
…
(starts compiling with xcodebuild)
…
  SOLINK_MODULE(target) Release/grpc_node.node
  COPY /…/grpc/src/node/extension_binary/grpc_node.node
  TOUCH Release/obj.target/action_after_build.stamp
…

$ 
```
I didn't get any errors and it succeeded in building.

**Comment by murgatroid99** [comment]
If you were trying to follow the quickstart, you skipped a step. It says to run `npm install` in the subdirectory `grpc/examples/node/dynamic_codegen`. That should install from npm instead of the local copy. If you want to build the local copy from source, you need to run `git submodule update --init` before running `npm install`.

**Comment by Martin-Pitt** [comment]
Oh yeah. That doesn't seem intuitive. Download source and examples, to checkout a very specific version and then to download another copy from somewhere else?

Why not host the examples in separate repos?
